lib/wallet/http.js

```diff
@@ -833,6 +783,405 @@ class HTTP extends Server {
       await req.wallet.resend();
       res.json(200, { success: true });
     });
+
+    // Wallet Name
+    this.get('/wallet/:id/name', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+
+      assert(name, 'Must pass name.');
+      assert(rules.verifyName(name), 'Must pass valid name.');
+
+      const height = this.wdb.height;
+      const network = this.network;
+      const ns = await req.wallet.getNameStateByName(name);
+
+      res.json(200, ns.getJSON(height, network));
+    });
+
+    // Wallet Names/Auctions
+    this.get('/wallet/:id/names', async (req, res) => {
+      const height = this.wdb.height;
+      const network = this.network;
+
+      const names = await req.wallet.getNames();
+      const items = [];
+
+      for (const ns of names)
+        items.push(ns.getJSON(height, network));
+
+      res.json(200, items);
+    });
+
+    // Wallet Auction
+    this.get('/wallet/:id/auction', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+
+      const height = this.wdb.height;
+      const network = this.network;
+
+      assert(name, 'Must pass name.');
+      assert(rules.verifyName(name), 'Must pass valid name.');
+
+      const ns = await req.wallet.getNameStateByName(name);
+      const bids = await req.wallet.getBidsByName(name);
+      const reveals = await req.wallet.getRevealsByName(name);
+
+      const info = ns.getJSON(height, network);
+      info.bids = [];
+      info.reveals = [];
```

Comment:
Not including opens or redeems or whatever else here?

Reply:
This is what is indexed:

https://github.com/handshake-org/hsd/blob/41ff9172ba32ce4b2af78dc11ec4045e0b37168e/lib/wallet/layout.js#L97-L104

Reply:
I'm going to skip returning anything else for now since it will dramatically increase the scope of this PR

Reply:
Just throwing this thought out there, but it would be nice to have a helper function that can group a Bid, Renewal, and Redeem into one object. That way you could just return that object, and it would encompass the 3 potential tx types. Likely out of scope for this PR, but wanted to suggest it anyway. 

Also I think only 1 open is valid per name now, so I think potentially returning that would be a good idea.

Reply:
I was originally thinking that to return the open, a function could be added to the wallet `getOpenByName`. This would fit in with the current way that this handler is architected. I don't think that the open would be indexed unless you opened it using the wallet. Would it be useful to go back and index the open for any name that you bid on? Then a service could know who opened the auction, which is valuable information when participating in an auction.

> it would be nice to have a helper function that can group a Bid, Renewal, and Redeem into one object

Could you elaborate on this more? What is the usecase? Is it to have a global view or a personal view? As in, do you want to be able to query for *all* transactions referring to an auction? Or just transactions that this particular wallet account sent pertaining the auction?